frame/nomination-pools/src/lib.rs

```diff
@@ -2033,10 +2055,10 @@ impl<T: Config> Pallet<T> {
 		who: &T::AccountId,
 	) -> Result<(PoolMember<T>, BondedPool<T>, RewardPool<T>), Error<T>> {
 		let member = PoolMembers::<T>::get(&who).ok_or(Error::<T>::PoolMemberNotFound)?;
-		let bonded_pool =
-			BondedPool::<T>::get(member.pool_id).defensive_ok_or(Error::<T>::PoolNotFound)?;
-		let reward_pool =
-			RewardPools::<T>::get(member.pool_id).defensive_ok_or(Error::<T>::PoolNotFound)?;
+		let err1: Error<T> = DefensiveError::PoolNotFound.into();
+		let err2: Error<T> = DefensiveError::PoolNotFound.into();
+		let bonded_pool = BondedPool::<T>::get(member.pool_id).defensive_ok_or(err1)?;
```

Comment:
This should work as well and does not need additional error variables.
```suggestion
		let bonded_pool = BondedPool::<T>::get(member.pool_id)
			.defensive_ok_or::<Error<T>>(DefensiveError::PoolNotFound.into())?;
```